include/pagmo/algorithms/gwo.hpp

```diff
@@ -136,19 +138,19 @@ class gwo
         m_log.clear();
 
         // Some vectors used during evolution are declared.
-        vector_double a_vector(3);          // vector coefficient for encircling prey
-        vector_double c_vector(3);          // vector coefficient for encircling prey
-        vector_double d_vector(3);          // position vector for alpha, beta and delta
-        vector_double x_vector(3);          // vector used to calculate position of an agent in a single dimension
-        double a;                           // coefficient which decrease linearly from 2 to 0 over generations
-        double r1, r2;                      // random coefficient between 0 and 1;
-        std::vector<std::size_t> index_vec; // used to stored sorted index
+        vector_double a_vector(3); // vector coefficient for encircling prey
+        vector_double c_vector(3); // vector coefficient for encircling prey
+        vector_double d_vector(3); // position vector for alpha, beta and delta
+        vector_double x_vector(3); // vector used to calculate position of an agent in a single dimension
+        double a;                  // coefficient which decrease linearly from 2 to 0 over generations
+        double r1, r2;             // random coefficient between 0 and 1;
         auto agents_position = pop.get_x();
         auto init_fit = pop.get_f();
-        for (unsigned int i = 0; i != init_fit.size(); ++i)
-            index_vec.push_back(i);
-        std::sort(index_vec.begin(), index_vec.end(),
-                  [&](unsigned int a, unsigned int b) { return init_fit[a][0] > init_fit[b][0]; });
+        std::vector<unsigned int> index_vec(init_fit.size()); // used to stored sorted index
+        std::iota(index_vec.begin(), index_vec.end(), 0);
```

Comment:
@michiboo when you use ``iota``, you must make sure that the initial value, 0 in this case, is of the correct type. The way you are using it here, the result of ``iota`` will be a signed ``int``, which is not what you want. You should write ``decltype(init_fit.size())(0)`` instead of just ``0``.